README.md
[![Project Status](https://www.repostatus.org/badges/latest/active.svg)](http://www.repostatus.org/#active) [![PyPI version](https://badge.fury.io/py/erichek.svg)](https://badge.fury.io/py/erichek) [![Python version](https://img.shields.io/pypi/pyversions/erichek.svg?style=plastic)](https://img.shields.io/pypi/pyversions/erichek.svg?style=plastic) [![PyPI version](https://img.shields.io/pypi/wheel/erichek.svg?style=plastic)](https://img.shields.io/pypi/pyversions/erichek.svg?style=plastic) [![Travis CI](https://travis-ci.org/Kristinita/Erichek.svg?branch=master)](https://travis-ci.org/Kristinita/Erichek) [![AppVeyor](https://ci.appveyor.com/api/projects/status/gscnxa6gj2oej1gn?svg=true)](https://ci.appveyor.com/project/Kristinita/erichek) [![Commitizen-friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg?style=plastic)](http://commitizen.github.io/cz-cli/) [![PePy](https://pepy.tech/badge/erichek)](https://pepy.tech/project/erichek)

Erichek — validator for [**Eric's rooms**](https://github.com/Kristinita/Erics-Green-Room).

See full description in **<https://Kristinita.github.io/Programs/Erichek>** page.


CHANGELOG.md
#### 0.4.0 (2019-05-06)

##### Refactors

* **python:** big refactoring ([4f7c3af3](https://github.com/Kristinita/Erichek/commit/4f7c3af3dc47694b16cf7edfc1a40adba95b9037))

#### 0.3.2 (2018-08-03)

##### Continuous Integration

* **ci:**  fix typos in config file ([555484f9](https://github.com/Kristinita/Erichek/commit/555484f9c58ef0ec8b2f606278bd87dbb04a26ce))

#### 0.3.1 (2018-08-03)

##### Continuous Integration

* **travis:**  Test CI after pyfancy publishing ([89050bb7](https://github.com/Kristinita/Erichek/commit/89050bb709bff6c44bf9e9dd07bcf69b99ae872a))

##### New Features

* **version:**  update ([94b3f4d3](https://github.com/Kristinita/Erichek/commit/94b3f4d3e15114b9062d6ce1b1090e62a2258656))

#### 0.2.1 (2018-02-09)

##### Bug Fixes

* **clize:**  Fix erichek --version ([2aedc8d0](https://github.com/Kristinita/Erichek/commit/2aedc8d0c414046e14f43b69fa4244ac4aa0051d))

##### Refactors

* **release-it:**  Add npm.publish false to configuration ([a44ea49f](https://github.com/Kristinita/Erichek/commit/a44ea49feb7c4e1bb45062fe419ffc826bc3dc8c))

##### Code Style Changes

* **badges:**  Make badges inline elements ([754b4550](https://github.com/Kristinita/Erichek/commit/754b4550f9583aef66946600d4f75de858532c29))
* **filenames:**
  *  Merge new filenames to master branch ([099731b7](https://github.com/Kristinita/Erichek/commit/099731b774e12cdb9456cea241596b3bdd8d6c13))
  *  README.rst, not README.RST ([760e2428](https://github.com/Kristinita/Erichek/commit/760e2428b87f055708af9ed178015d76a507f1a2))
  *  Push renamed files to GitHub ([588336d7](https://github.com/Kristinita/Erichek/commit/588336d72a91dfb1742a688390852ac22519b5d4))
  *  Capital letters for some filenames ([635fa180](https://github.com/Kristinita/Erichek/commit/635fa180e4cca95b38b92f06b018b9af1dc907d1))

#### 0.2.0 (2018-02-07)

##### Build System / Dependencies

* **wheel:**  Migrate to Wheel ([6ce8957b](https://github.com/Kristinita/Erichek/commit/6ce8957ba8bc4800b1f6ed622581cf769862b629))

##### Continuous Integration

* **circleci:**  Fix Circle CI build ([9fa65283](https://github.com/Kristinita/Erichek/commit/9fa65283b38fc5a53bf9a3fec5e7fafffa62324d))

##### Documentation Changes

* **badge:**  Add Wheel badge ([fe389db9](https://github.com/Kristinita/Erichek/commit/fe389db94c00b68229e842af0341a43efb4b4f72))

##### Bug Fixes

* **examples:**  Remove unexpected files ([bf92f432](https://github.com/Kristinita/Erichek/commit/bf92f4325004720b2a98c1a7f5522f745026a287))

#### 0.1.1 (2018-02-04)

##### Chores

* **badges:**  Add new badges ([79d861a9](https://github.com/Kristinita/Erichek/commit/79d861a9741bdcf14ed4fefcb95af64e23cb3ebe))

##### Code Style Changes

* **setup.cfg:**  Remove quotes from description variable ([af40ba04](https://github.com/Kristinita/Erichek/commit/af40ba045d13da5fa8687515b318a131f9da7017))

#### 0.1.0 (2018-02-04)

##### Build System / Dependencies

* **travisci, appveyor, circleci:**  Configure YAML files for CI ([bacc0513](https://github.com/Kristinita/Erichek/commit/bacc0513711a492058819890a768fee540529234))
* **release-it:**  Modify .release-it.json ([16fb0398](https://github.com/Kristinita/Erichek/commit/16fb0398d58847cbee5ebfc6854e1a4b7601a034))

##### Chores

* **erichek:**  Add License ([6cfd6102](https://github.com/Kristinita/Erichek/commit/6cfd61027b73af52c1ae15977ed95f70384f2ab1))

##### Continuous Integration

* **appveyor:**
  *  AppVeyor configuration only for Python 3 ([3469b623](https://github.com/Kristinita/Erichek/commit/3469b6231137c61bbff46145fa50e29b9cf9c30b))
  *  Fix typo ([387398ee](https://github.com/Kristinita/Erichek/commit/387398ee0d790cd129846b43ccd74006a0d9b3e9))
  *  Add AppVeyor support ([ac6b13ff](https://github.com/Kristinita/Erichek/commit/ac6b13ff2d8b9aeee676538755512c82502deb9a))
* **lintly:**  Add lintly.yml file ([cb4d8418](https://github.com/Kristinita/Erichek/commit/cb4d84188c021cd86d08439501ecf6d1f7553e98))

##### Documentation Changes

* **readme:**
  *  Add bold links ([99e9b0b1](https://github.com/Kristinita/Erichek/commit/99e9b0b19526b7dff12d45bb5243b44e209287fe))
  *  Fix Readme.rst ([4b682c3a](https://github.com/Kristinita/Erichek/commit/4b682c3a1728dff3290adc1aa61cf1158ecde7e8))

##### New Features

* **erichek:**  Initialization Erichek Python package ([9d819b2a](https://github.com/Kristinita/Erichek/commit/9d819b2aec9274c59ac1825a26967ff9dfa0b4b5))



erichek/__main__.py
# -*- coding: utf-8 -*-
# @Author: SashaChernykh
# @Date: 2018-01-22 07:32:05
# @Last Modified time: 2019-05-04 09:16:54
"""Run tests.

Main file for running Erichek tests.
"""
from clize import run

from erichek.eric_config import clize_log_level
from erichek.eric_config import pyfancy_error
from erichek.eric_config import pyfancy_notice
from erichek.eric_config import version

from erichek.eric_summary import eric_body_summary
from erichek.eric_summary import eric_encoding_summary
from erichek.eric_summary import eric_head_summary
from erichek.eric_summary import eric_regex_summary


# ASCII-art printing
# [BUG] ASCII-ART output, when --help or --version
# https://github.com/epsy/clize/issues/38#issue-290426111
# https://stackoverflow.com/a/9638532/5951529
# import sys
# from colorama import init
# from pyfiglet import figlet_format
# from termcolor import cprint

# strip colors if stdout is redirected
# init(strip=not sys.stdout.isatty())


def main():
    """Run Erichek.

    Run all modules of Erichek.
    If no errors — validation success,
    Else — exit(1).

    [NOTE] Do not use “exit=False” or “exit=True” in “run” function!
    https://github.com/epsy/clize/issues/33#issuecomment-354849918
    If “exit=True”, “erichek” will not work;
    else “exit=False”, all erichek modules will run, if “erichek --version” will run.
    """
    run(clize_log_level, alt=[version], exit=False)
    # If all instead of multiple if and:
    # https://stackoverflow.com/a/9504681/5951529
    if all([
            eric_encoding_summary(),
            eric_body_summary(),
            eric_head_summary(),
            eric_regex_summary()
    ]):
        pyfancy_notice(
            "Congratulations! You haven't errors in your packages!")
        # cprint(figlet_format('\nSuccess', font='starwars'),
        # 'white', 'on_green', attrs=['bold'])
    else:
        pyfancy_error(
            "You have errors in your packages. Please, fix them.")
        # cprint(figlet_format('\nFailure', font='starwars'),
        # 'yellow', 'on_red', attrs=['bold'])
        exit(1)


if __name__ == '__main__':
    main()
